src/App.js

```diff
@@ -7,19 +7,27 @@ import { petFilter } from "./services/PetService";
 
 const App = () => {
   const petTypes = ["All", "Cat", "Dog", "Other"];
-  const [pets] = useState(dataSource);
-  const [filteredPets, setFilteredPets] = useState(dataSource);
+  const initalPetTabs = {};
+  petTypes.forEach((type, index) => {
+    initalPetTabs[index] = type === "All" ? dataSource.AllPets : [];
+  });
+
+  const [pets] = useState(dataSource.AllPets);
+  const [petTabs, setPetTabs] = useState(initalPetTabs);
   const [selectedPetTypeIndex, setSelectedPetTypeIndex] = useState(0);
 
   const getFilteredPets = petTypeFilter =>
     petTypeFilter && petTypeFilter === "All"
       ? pets
-      : pets.filter(pet => petFilter(pet.species, petTypeFilter));
+      : pets.filter(pet => petFilter(pet.Species, petTypeFilter));
 
   useEffect(() => {
     const petType = petTypes[selectedPetTypeIndex];
     const filteredPets = getFilteredPets(petType);
-    setFilteredPets(filteredPets);
+    const updatedPetTabs = Object.assign({}, petTabs);
```

Comment:
move this to a handle event function since we don't need store the index in state.

Reply:
ended up needing the index still, but this works